### PR TITLE
feat: Add size props for Badge 

### LINKED ISSUE #25371: badge size (small)
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
badge is to large right now (for an icon)
image
https://codesandbox.io/s/antd-reproduction-template-lo61v?file=/index.js
### What does the proposed API look like?
`<badge size='small' /> `

### PR DESCRIPTION
**Pull request by moonrailgun**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #25371

### 💡 Background and solution

Add a size props for component Badge 

In some case, default size is too large

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Add new props `size` for Badge   |
| 🇨🇳 Chinese | Badge 新增 `size` 属性用于设置大小。   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/badge/demo/size.md](https://github.com/moonrailgun/ant-design/blob/badge-size/components/badge/demo/size.md)
[View rendered components/badge/index.en-US.md](https://github.com/moonrailgun/ant-design/blob/badge-size/components/badge/index.en-US.md)
[View rendered components/badge/index.zh-CN.md](https://github.com/moonrailgun/ant-design/blob/badge-size/components/badge/index.zh-CN.md)

### CODE REVIEW

**Comment by afc163** [comment]
feature branch please~

File: components/badge/demo/size.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## en-US
+
+Size of Badge.
+
+```jsx
+import { Badge } from 'antd';
+
+ReactDOM.render(
+  <div>
```
Comment (afc163) [inline components/badge/demo/size.md:20]:
```suggestion
  <>
```

File: components/badge/demo/size.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+order: 9
+title:
+  zh-CN: 大小
+  en-US: Size
+---
+
+## zh-CN
+
+对有数字的 Badge 的大小
```
Comment (afc163) [inline components/badge/demo/size.md:10]: 这句话没看懂。

Comment (moonrailgun) [reply components/badge/demo/size.md]: 已修改

File: components/badge/demo/size.md [hunk 3]
```diff
@@ -7,24 +7,24 @@ title:
 
 ## zh-CN
 
-对有数字的 Badge 的大小
+可以设置有数字徽标的大小
```
Comment (afc163) [inline components/badge/demo/size.md:10]: 标点。

File: components/badge/index.en-US.md [hunk 4]
```diff
@@ -24,6 +24,7 @@ Badge normally appears in proximity to notifications or user avatars with eye-ca
 | overflowCount | Max count to show | number | 99 |  |
 | showZero | Whether to show badge when `count` is zero | boolean | false |  |
 | status | Set Badge as a status dot | `success` \| `processing` \| `default` \| `error` \| `warning` | - |  |
+| size | If `count` is set, `size` sets the size of badge | `default` \| `small` | - |  |
```
Comment (xrkffgg) [inline components/badge/index.en-US.md:27]: Add version 4.6.0

File: components/badge/style/index.less [hunk 5]
```diff
@@ -32,6 +32,15 @@
     }
   }
 
+  &-count-sm {
```
Comment (xrkffgg) [inline components/badge/style/index.less:35]: 观察下 紧凑模式

Comment (moonrailgun) [reply components/badge/style/index.less:35]: 我要如何才能看到紧凑模式下的表现呢?

Comment (moonrailgun) [reply components/badge/style/index.less:35]: 紧凑模式下表现正常

File: components/style/themes/default.less [hunk 6]
```diff
@@ -645,8 +645,10 @@
 // Badge
 // ---
 @badge-height: 20px;
+@badge-height-sm: 16px;
```
Comment (afc163) [inline components/style/themes/default.less:648]: 区分度感觉不太够，改成 14px 如何？

Comment (moonrailgun) [reply components/style/themes/default.less:648]: image
image

可以，相比下来的确好像原来的区分度不够。

**Comment by afc163** [comment]
image

补充一下

**Comment by moonrailgun** [comment]
@afc163 已更新

**Review by afc163 — APPROVED** [review]

**Pull request merged**